### ISSUE TITLE
Carbon Components React: SideNavMenu component doesn't have WAI-ARIA role attribute.

### ISSUE THREAD

**Issue by AbrahamLara** [carbon-design-system/carbon#6940]
## Environment

> Operating system: **macOS Catalina**

> Browser: **Chrome**

> Automated testing tool and ruleset: **IBM Equal Access Accessibility Checker** / **W3C WAI-ARIA 1.1 Required Context Role**

> Assistive technology used to verify

## Detailed description

> What version of the Carbon Design System are you using? **7.10.3**

> What did you expect to happen? **Provide a WAI-ARIA role attribute**

> What happened instead? **The component does not have a WAI-ARIA role attribute**

> What WCAG 2.1 checkpoint does the issue violate? **WCAG 2.1 - SC 4.1.2 Name, Role, Value**

## Steps to reproduce the issue

1. Download [IBM Equal Access Accessibility Checker](https://chrome.google.com/webstore/detail/ibm-equal-access-accessib/lkcagbfjnkomcinoddgooolagloogehp) chrome extension
2. Visit https://7it9s.csb.app/
3. Open the menu in the side nav
4. Click on one of the links
5. Open chrome dev tools
6. Navigate to 'Accessibility Assessment' tab
7. Click 'Scan' - you should now see 2 violations

> Please create a reduced test case in CodeSandbox
>
> - React:
>  https://codesandbox.io/s/elastic-water-7it9s?fontsize=14&hidenavigation=1&theme=dark

## Additional information

<img width="1439" alt="Screen Shot 2020-09-30 at 4 25 02 PM" src="https://user-images.githubusercontent.com/37315075/94736035-9d2e2b80-0339-11eb-8137-098feb75ab52.png">

I decided to check the latest version of carbon-components-react to see if the component can be given a role or aria-label prop. This didn't seem to be the case so I decided to create this issue.

**Comment by jannaee** [comment]
I'd like to take a look into this for Hacktober fest. Could you assign it to me?
